tests/functional/build/test_build.py

```diff
@@ -0,0 +1,296 @@
+        "model_99.sql": models_failing__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def seeds():
+    return {"countries.csv": seeds__countries_csv}
+
+
+@pytest.fixture(scope="class")
+def models_circular_relationship():
+    return {
+        "test.yml": models_circular_relationship__test_yml,
+        "model_0.sql": models_circular_relationship__model_0_sql,
+        "model_1.sql": models_circular_relationship__model_1_sql,
+        "model_99.sql": models_circular_relationship__model_99_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models_simple_blocking():
+    return {
+        "schema.yml": models_simple_blocking__schema_yml,
+        "model_b.sql": models_simple_blocking__model_b_sql,
+        "model_a.sql": models_simple_blocking__model_a_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def models_interdependent():
+    return {
+        "schema.yml": models_interdependent__schema_yml,
+        "model_c.sql": models_interdependent__model_c_sql,
+        "model_a.sql": models_interdependent__model_a_sql,
+    }
+
+
+@pytest.fixture(scope="class")
+def project_files(
+    project_root,
+    snapshots,
+    tests_failing,
+    models,
+    models_failing,
+    seeds,
+    models_circular_relationship,
+    models_simple_blocking,
+    models_interdependent,
+):
+    write_project_files(project_root, "snapshots", snapshots)
+    write_project_files(project_root, "tests-failing", tests_failing)
+    write_project_files(project_root, "models", models)
+    write_project_files(project_root, "models-failing", models_failing)
+    write_project_files(project_root, "seeds", seeds)
+    write_project_files(project_root, "models-circular-relationship", models_circular_relationship)
+    write_project_files(project_root, "models-simple-blocking", models_simple_blocking)
+    write_project_files(project_root, "models-interdependent", models_interdependent)
+
+
+class BuildBase:
+    def build(self, expect_pass=True, extra_args=None, **kwargs):
+        args = ["build"]
+        if kwargs:
+            args.extend(("--args", yaml.safe_dump(kwargs)))
+        if extra_args:
+            args.extend(extra_args)
+
+        return run_dbt(args, expect_pass=expect_pass)
+
+
+class TestPassingBuild(BuildBase):
+    @pytest.fixture(scope="class")
+    def model_path(self):
+        return "models"
+
+    def test_build_happy_path(
+        self,
+        project,
+    ):
+        self.build(expect_pass=True)
+
+
+class TestFailingBuild(BuildBase):
+    @pytest.fixture(scope="class")
+    def models(self, models_failing):  # noqa: F811
+        return {
+            "model_3.sql": models_failing["model_3.sql"],
+            "model_2.sql": models_failing["model_2.sql"],
+            "test.yml": models_failing["test.yml"],
+            "model_0.sql": models_failing["model_0.sql"],
+            "model_1.sql": models_failing["model_1.sql"],
+            "model_99.sql": models_failing["model_99.sql"],
+        }
```

Comment:
Here's the example of how to define the models without everything at the top of this file.

```
def models(self):
    return {
        "model_3.sql": models_failing__model_3_sql,
        "model_2.sql": models_failing__model_2_sql,
        "test.yml": models_failing__test_yml,
        "model_0.sql": models_failing__model_0_sql,
        "model_1.sql": models_failing__model_1_sql,
        "model_99.sql": models_failing__model_99_sql,
    }
```

Reply:
When you import the strings this would be how you define the models fixture and can then remove the `# noqa`.